File: BmiCalc/app/src/main/java/pamo/bmicalc/ui/calculate/CalculateViewModel.java
package pamo.bmicalc.ui.calculate;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class CalculateViewModel extends ViewModel {
    public MutableLiveData<Map<String, String>> stringDataStore = new MutableLiveData<>();
    public Map<String, String> stringDataMap = new HashMap<>();

    public CalculateViewModel() {
        stringDataStore.setValue(stringDataMap);
    }

    public String getStringMapValueByKey(String key){
        return Objects.requireNonNull(stringDataStore.getValue()).get(key);
    }

    public void addStringDataToMap(String key, String value){
        Objects.requireNonNull(stringDataStore.getValue()).put(key, value);
    }
}

File: BmiCalcKotlin/app/src/main/java/pamo/bmicalc_kotlin/ui/calculate/CalculateViewModel.kt
package pamo.bmicalc_kotlin.ui.calculate

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.util.HashMap

class CalculateViewModel : ViewModel() {
    private var stringDataStore = MutableLiveData<MutableMap<String, String>>()
    private var stringDataMap: MutableMap<String, String> = HashMap()

    init {
        stringDataStore.value = stringDataMap
    }

    fun getStringMapValueByKey(key: String): String? {
        return stringDataStore.value!![key]
    }

    fun addStringDataToMap(key: String, value: String) {
        stringDataStore.value!![key] = value
    }
}